src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -87,28 +115,50 @@ public String getTagValue() {
       return Long.toString(tagControl.getTagValueAsLong());
    }
 
-   public void recordTagValue(String time)
+   /**
+    * Record tag value will be called after parsing the historical log values.
+    * Historical logs are added to internal queue through this function.
+    *
+    * @param time is the time in historical logs in epoch time seconds
+    * (Referred to as timeInt in the EBD generated file)
+    * @param value is the value in historical logs
+    */
+   public void recordTagValue(String time, String value)
    {
+      // convert time string holding a epoch time in seconds to timestamp
+      long epochSeconds = Long.parseLong(time);
+      long epochTime = epochSeconds*1000;

```

Comment:
Maybe make this more clear that this is milliseconds as you have epochSeconds and epochTime